stackslib/src/net/relay.rs

```diff
@@ -528,6 +529,41 @@ impl Relayer {
         Ok(())
     }
 
+    /// Given Nakamoto blocks pushed to us, verify that they correspond to expected block data.
+    pub fn validate_nakamoto_blocks_push(
+        conn: &SortitionDBConn,
+        blocks_data: &NakamotoBlocksData,
+    ) -> Result<(), net_error> {
+        for NakamotoBlocksDatum(consensus_hash, block) in blocks_data.blocks.iter() {
+            let block_hash = block.header.block_hash();
+
+            // is this the right Stacks block for this sortition?
+            let sn = match SortitionDB::get_block_snapshot_consensus(conn.conn(), consensus_hash)? {
+                Some(sn) => {
+                    if !sn.pox_valid {
+                        info!( "Pushed block from consensus hash {consensus_hash} corresponds to invalid PoX state",);
+                        continue;
+                    }
+                    sn
+                }
+                None => {
+                    // don't know about this yet
+                    continue;
+                }
+            };
+
+            if !sn.sortition || sn.winning_stacks_block_hash != block_hash {
+                info!("No such sortition in block with consensus hash {consensus_hash}");
+
+                // TODO: once PoX is implemented, this can be permitted if we're missing the reward
```

Comment:
This comment is no longer true, since PoX is obviously implemented ;)

Reply:
Right -- in Nakamoto, this should be a `warn` case, and an `InvalidMessage` error, which here just means removing this comment and changing `info` to `warn`.